plugins/message_tracker.py

```diff
@@ -186,7 +186,7 @@ async def select_fetch_task(session: sqlalchemy.ext.asyncio.AsyncSession, subscr
                     sqlalchemy.null().label("before_snowflake"))
                 .join(ChannelState.channel)
                 .where(Channel.reachable, ChannelState.subscriber.in_(subs),
-                    ChannelState.earliest_thread_archive_ts != None)
+                    ChannelState.earliest_thread_archive_ts is not None)
```

Comment:
Have you tested this at all? This does not work. `ChannelState.earliest_thread_archive_ts` is a `sqlalchemy.Column` object that overrides `__ne__` so that `column != x` is translated into an SQL condition that gets added to the select. If you do `column is not x` it will actually compute on the python side (as `is`/`is not` cannot be overloaded in python), and this will be basically `select.where(False)` which will select nothing.